pkg/controller/certificates/internal/secretsmanager/secret.go

```diff
@@ -163,6 +191,11 @@ func (s *SecretsManager) setValues(crt *cmapi.Certificate, secret *corev1.Secret
 		secret.Data[cmmeta.TLSCAKey] = data.CA
 	}
 
+	annotations, err := SecretCertificateAnnotations(crt, data)
+	if err != nil {
+		return err
```

Comment:
suggestion: give more context on this error. If we don't give more context, the error that will show in the logs will feel a bit cryptic (for example, a failure to decode the certificate).

Reply:
Added some more context to there error, let me know if it need some more